ts/features/idpay/unsubscription/screens/UnsubscriptionConfirmationScreen.tsx

```diff
@@ -19,33 +20,57 @@ import { emptyContextualHelp } from "../../../../utils/emptyContextualHelp";
 import { useLegacyIOBottomSheetModal } from "../../../../utils/hooks/bottomSheet";
 import { UnsubscriptionCheckListItem } from "../components/UnsubscriptionCheckListItem";
 import { useUnsubscriptionMachineService } from "../xstate/provider";
-import { isLoadingSelector, selectInitiativeName } from "../xstate/selectors";
+import {
+  isLoadingSelector,
+  selectInitiativeName,
+  selectInitiativeType
+} from "../xstate/selectors";
 
-const unsubscriptionChecks: ReadonlyArray<{ title: string; subtitle: string }> =
-  [
-    {
-      title: I18n.t("idpay.unsubscription.checks.1.title"),
-      subtitle: I18n.t("idpay.unsubscription.checks.1.content")
-    },
-    {
-      title: I18n.t("idpay.unsubscription.checks.2.title"),
-      subtitle: I18n.t("idpay.unsubscription.checks.2.content")
-    },
-    {
-      title: I18n.t("idpay.unsubscription.checks.3.title"),
-      subtitle: I18n.t("idpay.unsubscription.checks.3.content")
-    },
-    {
-      title: I18n.t("idpay.unsubscription.checks.4.title"),
-      subtitle: I18n.t("idpay.unsubscription.checks.4.content")
-    }
-  ];
+const refundUnsubscriptionChecks: ReadonlyArray<{
+  title: string;
+  subtitle: string;
+}> = [
+  {
+    title: I18n.t("idpay.unsubscription.checks.1.title"),
+    subtitle: I18n.t("idpay.unsubscription.checks.1.content")
+  },
+  {
+    title: I18n.t("idpay.unsubscription.checks.2.title"),
+    subtitle: I18n.t("idpay.unsubscription.checks.2.content")
+  },
+  {
+    title: I18n.t("idpay.unsubscription.checks.3.title"),
+    subtitle: I18n.t("idpay.unsubscription.checks.3.content")
+  },
+  {
+    title: I18n.t("idpay.unsubscription.checks.4.title"),
+    subtitle: I18n.t("idpay.unsubscription.checks.4.content")
+  }
+];
+
+const discountUnsubscriptionChecks: ReadonlyArray<{
+  title: string;
+  subtitle: string;
+}> = [
+  {
+    title: I18n.t("idpay.unsubscription.checks.1.title"),
+    subtitle: I18n.t("idpay.unsubscription.checks.1.content")
+  },
+  {
+    title: I18n.t("idpay.unsubscription.checks.3.title"),
+    subtitle: I18n.t("idpay.unsubscription.checks.3.content")
+  }
+];
 
 const UnsubscriptionConfirmationScreen = () => {
   const machine = useUnsubscriptionMachineService();
-
+  const initiativeType = useSelector(machine, selectInitiativeType);
   const isLoading = useSelector(machine, isLoadingSelector);
   const initiativeName = useSelector(machine, selectInitiativeName);
+  const unsubscriptionChecks =
+    initiativeType === InitiativeRewardTypeEnum.DISCOUNT
+      ? discountUnsubscriptionChecks
+      : refundUnsubscriptionChecks;
```

Comment:
What do you think about moving this logic to a specific selector? Maybe a `selectChecks` selector which returns the right checks to show based on the `initiativeType `